biojava-structure/src/main/java/org/biojava/nbio/structure/io/mmcif/SimpleMMcifConsumer.java

```diff
@@ -920,6 +929,28 @@ public void documentEnd() {
 		}
 
 	}
+	
+	/**
+	 * Ensure that all the alt loc groups have all the atoms in the main group
+	 */
+	private void correctAltLocGroups() {
+		for (int i =0; i< structure.nrModels() ; i++){
+			for (Chain chain : structure.getModel(i)) {
+				for (Group group : chain.getAtomGroups()) {
+					for (Group altLocGroup : group.getAltLocs()) { 
+						for ( Atom groupAtom : group.getAtoms()) {
+							// If this alt loc doesn't have this atom
+							if (! altLocGroup.hasAtom(groupAtom.getName())) {
+								if (altLocGroup.getPDBName().equals(group.getPDBName())) {
+									altLocGroup.addAtom(groupAtom);
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	}
```

Comment:
This block is a duplicate of one above in PdbFileParser. We should extract it to some utility class.